dpe/src/commands/destroy_context.rs

```diff
@@ -33,7 +33,7 @@ impl CommandExecution for DestroyCtxCmd {
         // contexts_to_destroy[i] == true implies dpe.contexts[i] must be destroyed
         let mut contexts_to_destroy = [false; MAX_HANDLES];
         // updated_children[i] holds i's non-destroyed children.
-        let mut updated_children = [0u32; MAX_HANDLES];
+        let mut updated_children = [u32::MAX; MAX_HANDLES];
```

Comment:
I think this code works correctly, but is a bit hard to follow, which could lead to issues being introduced in the future. I'll think about this a little more, but maybe we could try to refactor this to simplify it.

Reply:
Here's a straw proposal for how this could maybe be changed:

```rust
// TODO: Is there a way to be extra sure this is being copied and not moved?
// Like calling a .copy function?
let mut tmp_contexts = dpe.contexts;

let to_destroy = (1 << idx) | dpe.get_descendants(context)?;

for idx, c in tmp_contexts {
    // Clears all the to_destroy bits in the children of every context
    c.children &= ~to_destroy;
    if to_destroy & (1 << idx) != 0 {
        c.destroy();
    }
}

dpe.contexts = tmp_contexts;
```

I _think_ this does the same thing as the existing code.

Looking at this, I'm not even sure we need to do the copying because I don't think any of these calls can fail.

Reply:
Thanks, this is way simpler. Also removed the copying.